### PR TITLE
Add k8 (rgb) converted to info.json style

### PR DESCRIPTION
**Pull request by tbowmo**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tbowmo** [comment]
Tested with K8 rgb ISO variant.

All other variants is verified by a small python test routine comparing the autogenerated mapping macro, with the one in the original header files (from old master)

**Comment by davecarrijo** [comment]
Is via working?

**Comment by tbowmo** [comment]
Should be, but haven't tested yet. I only did a quick test of plain iso layout, which worked as expected.

**Comment by davecarrijo** [comment]
I have the k8 ansi variant, if you need a tester I will gladly help

**Comment by tbowmo** [comment]
Via is now verified as working (iso variant). Also updated default via keymap to be the same as windows iso layout.

File: keyboards/keychron/k8/rgb/config.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 17
+
+// Connects each switch in the dip switch to the GPIO pin of the MCU
+#define DIP_SWITCH_PINS { D4, D5 }
+
+/* LED Status indicators */
+#define LED_CAPS_LOCK_PIN B12
+#define LED_PIN_ON_STATE  1
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/config.h]: this is the default value, safe to omit

File: keyboards/keychron/k8/rgb/keymaps/default/rules.mk [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# https://beta.docs.qmk.fm/developing-qmk/qmk-reference/getting_started_make_guide#rules-mk-options
+
+# AUTO_SHIFT_ENABLE = yes
+# MOUSEKEY_ENABLE = yes
+EXTRAKEY_ENABLE = yes    # Audio control and System control
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/keymaps/default/rules.mk]: why is this here?

File: keyboards/keychron/k8/rgb/config.h (hunk 1 above)
Comment (dexter93) [inline keyboards/keychron/k8/rgb/config.h]: this can go. it's the default value

File: keyboards/keychron/k8/rgb/ansi/config.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// RGB Matrix Config
+#define DRIVER_LED_TOTAL 87
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/ansi/config.h]: these can now be 1 define

File: keyboards/keychron/k8/rgb/config.h (hunk 1 above)
Comment (tbowmo) [reply keyboards/keychron/k8/rgb/config.h]: Gone it is..

File: keyboards/keychron/k8/rgb/optical_iso/rules.mk [hunk 4]
```diff
@@ -0,0 +1 @@
+OPT_DEFS += -DOPTICAL_MATRIX
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/optical_iso/rules.mk]: need to figure out what will happen with this. we need something that works against the main qmk scan now

File: keyboards/keychron/k8/rgb/keymaps/default/rules.mk (hunk 2 above)
Comment (tbowmo) [reply keyboards/keychron/k8/rgb/keymaps/default/rules.mk]: Because I didn't look through the keymap files ;) But it's removed, as it's in the base info.json file

File: keyboards/keychron/k8/rgb/optical_iso/rules.mk (hunk 4 above)
Comment (tbowmo) [reply keyboards/keychron/k8/rgb/optical_iso/rules.mk]: I left it here, as I was unsure and do not have an optical keyboard..

File: keyboards/keychron/k8/rgb/rules.mk [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# Build Options
+#   comment out to disable the options.
+#
+
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/rules.mk]: this is getting picked up by the linter and failing
```☒ Lint check failed for: keychron/k8/rgb```
remove the rules.mk from here and it will shut up

File: keyboards/keychron/k8/rgb/info.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": false,
+        "nkro": true,
+        "dip_switch_enable": true
+    },
+    "rgb_matrix": {
+        "driver": "SN32F24xB"
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/info.json]: does this work?

Comment (tbowmo) [reply keyboards/keychron/k8/rgb/info.json]: Yup, it does show up in the autogenerated rules.mk file:
`RGB_MATRIX_ENABLE ?= yes`

File: keyboards/keychron/k8/rgb/rules.mk (hunk 5 above)
Comment (tbowmo) [reply keyboards/keychron/k8/rgb/rules.mk]: bugger.. of course.. Why didn't I think of that :laughing:

File: keyboards/keychron/k8/rgb/config.h [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+/* key matrix size */
+#define MATRIX_ROWS 6
+#define MATRIX_COLS 17
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/config.h]: these can also go

Comment (tbowmo) [reply keyboards/keychron/k8/rgb/config.h]: Are you sure? They are used multiple places in config.h. The question is if the other places (where these are used) could be deleted as well?

Comment (tbowmo) [reply keyboards/keychron/k8/rgb/config.h]: Looking through the autogenerated header files, it seems that qmk figures this out from the json file.. So yes, they can be removed

File: keyboards/keychron/k8/rgb/info.json (hunk 6 above)
Comment (dexter93) [reply keyboards/keychron/k8/rgb/info.json]: somehow it doesn't get picked up here. weird

**Comment by davecarrijo** [comment]
so... with this we can use VIA now?

**Comment by tbowmo** [comment]
@davecarrijo seems that via is temporarily broken in this PR (it worked initially, but a lot has changed since then)

**Comment by davecarrijo** [comment]
> @davecarrijo seems that via is temporarily broken in this PR (it worked initially, but a lot has changed since then)

Did you changed the pid and the vid? What it changes?

**Comment by tbowmo** [comment]
I changed pid to the correct one for the iso variants, which also happens to fix via support :)

File: keyboards/keychron/k8/rgb/optical_iso/rules.mk (hunk 4 above)
Comment (dexter93) [reply keyboards/keychron/k8/rgb/optical_iso/rules.mk]: Probably going to go with https://github.com/SonixQMK/qmk_firmware/pull/299 once it gets the green light from qmk

File: keyboards/keychron/k8/rgb/info.json (hunk 6 above)
Comment (vware) [reply keyboards/keychron/k8/rgb/info.json]: obj_keychron_k8_rgb_iso/src/info_rules.mk:RGB_MATRIX_ENABLE ?= yes

**Comment by vware** [comment]
this is looking quite nice actually, and it is the basis for the bluetooth pr, any chance this will get merged soonish? ;)

**Comment by davecarrijo** [comment]
> I changed pid to the correct one for the iso variants, which also happens to fix via support :) 

The man !!!🫡

**Comment by davecarrijo** [comment]
> this is looking quite nice actually, and it is the basis for the bluetooth pr, any chance this will get merged soonish? ;)

That would made it just one of the best keyboards out there.

**Review by davecarrijo — COMMENTED** [review]
nice

File: .gitmodules [hunk 8]
```diff
@@ -4,8 +4,8 @@
 	branch = master
 [submodule "lib/chibios-contrib"]
 	path = lib/chibios-contrib
-	url = https://github.com/qmk/ChibiOS-Contrib
-	branch = master
+	url = https://github.com/SonixQMK/ChibiOS-Contrib
```
Comment (dexter93) [inline .gitmodules]: no longer required. please remove modifications on .gitmodules

File: keyboards/keychron/k8/rgb/optical_iso/rules.mk (hunk 4 above)
Comment (dexter93) [reply keyboards/keychron/k8/rgb/optical_iso/rules.mk]: this has now been updated in https://github.com/qmk/qmk_firmware/pull/19053
code is in upstream now, see the PR for usage refs

File: keyboards/keychron/k8/rgb/keymaps/via/k8_via_ansi.json [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+        "SOLID_MULTISPLASH",
+        1
+      ]
+    ],
+    "supportedLightingValues": [
+      128,
+      129,
+      131
+    ]
+  },
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/keymaps/via/k8_via_ansi.json]:
```suggestion
  "keycodes": ["qmk_lighting"],
  "menus": ["qmk_rgb_matrix"],
```
This will make it work with via v3 defs

File: keyboards/keychron/k8/rgb/keymaps/via/k8_via_ansi_optical.json [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+        "SOLID_MULTISPLASH",
+        1
+      ]
+    ],
+    "supportedLightingValues": [
+      128,
+      129,
+      131
+    ]
+  },
```
Comment (dexter93) [inline keyboards/keychron/k8/rgb/keymaps/via/k8_via_ansi_optical.json]:
```suggestion
  "keycodes": ["qmk_lighting"],
  "menus": ["qmk_rgb_matrix"],
```
This will make it work with via v3 defs

**Review by dexter93 — CHANGES_REQUESTED** [review]
suggestions for via v3 lighting
